frame/contracts/src/migration/v10.rs

```diff
@@ -117,7 +117,7 @@ pub struct Migration<T: Config> {
 type ContractInfoOf<T: Config> =
 	StorageMap<Pallet<T>, Twox64Concat, <T as frame_system::Config>::AccountId, ContractInfo<T>>;
 
-impl<T: Config> Migrate for Migration<T> {
+impl<T: Config> MigrationStep for Migration<T> {
```

Comment:
Continuing [this](https://github.com/paritytech/substrate/pull/14045#discussion_r1212023689) discourse: 

https://github.com/paritytech/substrate/blob/dbb20d3accc0331eb40af0c842427bfeeb430240/frame/contracts/src/migration/v10.rs#L163-L166

Do we really need to subtract `min_balance` here? Possible arguments versus are:
1. if `old_deposit > reducible_balance` then only `reducible_balance` would be charged here.
2. Looks like we already subtract it when calculate the deposit due:
https://github.com/paritytech/substrate/blob/6a89133090d77002224d3aa865aa85f57e835acd/frame/contracts/src/storage.rs#L120-L124
3. We are transferring from regular (not contract) account here. Tombstone-ing it is not end of the world, as it's holder could top-up it and bring it back to life again.

Reply:
1. We substract the min_balance because it was accounted into the deposit to account for the account creation, we would otherwise hit the fallback which is to transfer the reducible_balance.
2. We substract it here cause we want the total_deposit minus the ED that was added for the account

https://github.com/paritytech/substrate/blame/aeb24c2545422b1157bd6a85f4017aaf5287678a/frame/contracts/src/storage/meter.rs#L427-L428

I tested this back then when I made the original PR, I can try to test it again, it's a bit tricky now, cause I need to download a full snapshot to run try-runtime on it again, since v11 has been passed quite some blocks ago now

Reply:
Double tested it today, and indeed not substracting the min_balance makes you go to the fallback